compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -289,6 +289,34 @@ static int64_t getVectorSize(mlir::FunctionOpInterface entryPointFn,
   return getVectorSize(entryPointFn, byteWidth);
 }
 
+/// Returns true if the operation is a GenericOp implementing a supported
+/// transposition.
+static bool isSupportedTransposeOp(linalg::GenericOp genericOp) {
+  // Check that the op has at least 2 dimensions.
+  if (genericOp.getNumLoops() < 2) {
+    return false;
+  }
+
+  // Check that the op has only one input and one output.
+  // TODO(diegocaballero): Generalize to multiple inputs.
+  if ((genericOp.getNumDpsInputs() != 1) || (genericOp.getNumDpsInits() != 1)) {
+    return false;
+  }
+
+  // Check that all the iterators are parallel.
+  if (genericOp.getNumParallelLoops() != genericOp.getNumLoops()) {
+    return false;
+  }
+
+  // Check that the two indexing maps are a permutation of each other.
```

Comment:
The code seems to implement `Both indexing maps are permutations, and exactly one of them is the identity`. What does this comment mean by "are a permutation of each other" ?

Also, this condition could be simplified to `m0.isPermutation() && m1.isPermutation() && (m0.isIdentity() ^ m1.isIdentity())`.

What role is played by the separate checks for `isEmpty` ?

Reply:
As same as above response, the code is quite old. I can help update it either in the PR or in a follow-up. `isEmpty()` is the check for scalars. We don't need the specialized transpose lowering when they are scalars.